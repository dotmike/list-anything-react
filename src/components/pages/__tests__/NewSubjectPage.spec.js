import React from "react";
import configureStore from "redux-mock-store";

import ConnectedNewSubjectPage, {
  UnconnectedNewSubjectPage
} from "../NewSubjectPage";

const defaultProps = {
  history: {
    push: () => {}
  }
};

describe("ConnectedNewSubjectPage", () => {
  const mockStore = configureStore();
  const initialState = {};
  const props = { ...defaultProps, store: mockStore(initialState) };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(<ConnectedNewSubjectPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("UnconnectedNewSubjectPage", () => {
  const props = {
    ...defaultProps,
    createSubject: () => Promise.resolve()
  };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(
      <UnconnectedNewSubjectPage {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
