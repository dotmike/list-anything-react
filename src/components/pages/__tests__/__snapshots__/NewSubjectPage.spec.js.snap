// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedNewSubjectPage renders correctly 1`] = `
<NewSubjectPage
  createSubject={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`UnconnectedNewSubjectPage renders correctly 1`] = `
<Segment
  style={
    Object {
      "margin": "10px auto",
      "maxWidth": 500,
    }
  }
>
  <Header
    as="h2"
    color="teal"
    textAlign="center"
  >
    Create new Subject
  </Header>
  <NewSubjectForm
    submit={[Function]}
  />
</Segment>
`;
